src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -544,6 +545,14 @@ void CoreWorkerDirectTaskSubmitter::RequestNewWorkerIfNeeded(
       is_selected_based_on_locality);
   scheduling_key_entry.pending_lease_requests.emplace(task_id, *raylet_address);
   ReportWorkerBacklogIfNeeded(scheduling_key);
+
+  // Lease more workers if there are still pending tasks and
+  // and we haven't hit the max_pending_lease_requests yet.
+  if (scheduling_key_entry.task_queue.size() >
```

Comment:
Actually why do we need this logic? IIRC, we should already have the logic to limit # of task submission based on `max_pending_lease_requests`. Is there any reason why we need additional logic here?

Reply:
because the max_pending_lease_requests might change to a bigger number if the number of node increased.